test/core/surface/channel_init_test.cc

```diff
@@ -157,8 +166,36 @@ TEST(ChannelInitTest, CanAddBeforeAllOnce) {
   b.RegisterFilter(GRPC_CLIENT_CHANNEL, FilterNamed("foo")).BeforeAll();
   b.RegisterFilter(GRPC_CLIENT_CHANNEL, FilterNamed("bar"));
   b.RegisterFilter(GRPC_CLIENT_CHANNEL, FilterNamed("baz"));
+  b.RegisterFilter(GRPC_CLIENT_CHANNEL, FilterNamed("aaa")).Terminal();
   EXPECT_EQ(GetFilterNames(b.Build(), GRPC_CLIENT_CHANNEL, ChannelArgs()),
-            std::vector<std::string>({"foo", "bar", "baz"}));
+            std::vector<std::string>({"foo", "bar", "baz", "aaa"}));
+}
+
+TEST(ChannelInitTest, CanAddBeforeAllTwice) {
+  ChannelInit::Builder b;
+  b.RegisterFilter(GRPC_CLIENT_CHANNEL, FilterNamed("foo")).BeforeAll();
+  b.RegisterFilter(GRPC_CLIENT_CHANNEL, FilterNamed("bar")).BeforeAll();
+  b.RegisterFilter(GRPC_CLIENT_CHANNEL, FilterNamed("baz"));
+  b.RegisterFilter(GRPC_CLIENT_CHANNEL, FilterNamed("aaa")).Terminal();
+  EXPECT_DEATH_IF_SUPPORTED(b.Build(), "Unresolvable graph of channel filters");
+}
+
+TEST(ChannelInitTest, CanPostProcessFilters) {
+  ChannelInit::Builder b;
+  b.RegisterFilter(GRPC_CLIENT_CHANNEL, FilterNamed("foo"));
+  b.RegisterFilter(GRPC_CLIENT_CHANNEL, FilterNamed("aaa")).Terminal();
+  int called_post_processor = 0;
+  b.RegisterPostProcessor(
+      GRPC_CLIENT_CHANNEL,
+      ChannelInit::PostProcessorSlot::kXdsChannelStackModifier,
+      [&called_post_processor](ChannelStackBuilder& b) {
+        ++called_post_processor;
+        b.mutable_stack()->push_back(FilterNamed("bar"));
+      });
+  auto init = b.Build();
+  EXPECT_EQ(called_post_processor, 0);
```

Comment:
Shouldn't this be 1?

Reply:
At this point channel init has been built but not a channel - so we should not have called the post processor